### PR TITLE
task/02-interfaces

### PR DESCRIPTION
**Pull request by 3apy6a**
- [x] Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Interfaces for Product and Category based on [swagger](https://ma-backend-api.mocintra.com/docs#/categories/CategoriesController_getAll) are created and exported in separate file.
- [x]  Static header and footer are created according to [mobile](https://www.figma.com/file/mRAOMdJiAsD8fHSVnYwUl9/React-web-page?type=design&node-id=13-17490&mode=design&t=3Thf45kJJYMoxsc0-0) and [desktop](https://www.figma.com/file/MIKTcyr6SvspkZnIQ5nEPD/MA-Courses-2022-2023?type=design&node-id=2471-2253&mode=design&t=4DJ0BIdFZZ4mUgAn-0) designs.

Acceptance criteria for 1*

- [x]  Shared component SocialIcon is created and used for footer icons
- [x]  Link to social network and path to icon are passed to component by the typed props

[лючшій сайт](https://3apy6a.github.io/fe-react-2024/)

### CODE REVIEW

File: src/interfaces/Product.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type { Category } from '@/interfaces/Category.ts';
+import type { Image } from '@/interfaces/Image.ts';
+
+export interface Product {
+    id: number;
+    title: string;
+    price: number;
+    description: string;
+    images: Image[];
+    creationAt: string;
```
Comment (vadym-zinchenko-moc) [inline src/interfaces/Product.ts]: it's Date type. Check pls all Date types.

File: src/components/header/components/shared/icons/SignUpIcon.component.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import type { CustomClasses } from '@/interfaces/CustomClasses.ts';
+
+export const SignUpIconComponent = ({ customClasses }: CustomClasses) => (
+    <svg className={customClasses} width="21" height="17" viewBox="0 0 21 17" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (vadym-zinchenko-moc) [inline src/components/header/components/shared/icons/SignUpIcon.component.tsx:4]: You can export svg files from a figma-like file. And use with import. You can check how use others students.

File: src/components/header/components/desktop/Desktop.component.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import { LogoComponent } from '@/components/header/components/shared/Logo.component.tsx';
+
+import styles from './desktop.module.css';
+export const DesktopComponent = () => {
+    const wrapperClasses = classNames({
+        [styles.flex]: true,
+        [styles.align_center]: true,
+    });
+
+    const linkClasses = classNames({
```
Comment (vadym-zinchenko-moc) [inline src/components/header/components/desktop/Desktop.component.tsx:18]: very strange construction. What is all this for?

File: src/components/header/components/desktop/desktop.module.css [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  list-style: none;
+  margin: 0;
+  padding: 0;
+}
+
+.align_center {
+  align-items: center;
+}
+
+.color_white {
```
Comment (vadym-zinchenko-moc) [inline src/components/header/components/desktop/desktop.module.css:23]: for future need to read BEM methodology for class naming 
[https://en.bem.info/methodology/quick-start/](url)

File: package.json [hunk 5]
```diff
@@ -12,8 +12,10 @@
     "prepare": "husky install"
   },
   "dependencies": {
+    "classnames": "^2.5.1",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-responsive": "^10.0.0"
```
Comment (vadym-zinchenko-moc) [inline package.json:18]: I think we don't need to use any third-party libraries for such a simple functionality

File: src/components/header/components/shared/Cart.component.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+export const CartComponent = () => (
+    <a href={'/buy'}>
```
Comment (vadym-zinchenko-moc) [inline src/components/header/components/shared/Cart.component.tsx]: `href={'/buy'}`

I think you can use all `href` like this `href='/buy'`, without extra `{}`

File: src/components/header/components/mobile/Mobile.component.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+export const MobileComponent = () => (
+    <header className={styles.header}>
+        <LogoComponent />
+        <ul className={styles.menu}>
+            <li>
+                <CartComponent />
+            </li>
+            <li>
+                <a href={'#'}>
```
Comment (vadym-zinchenko-moc) [inline src/components/header/components/mobile/Mobile.component.tsx]: need to fix linter. Change to `href="/"`

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

**Comment by vadym-zinchenko-moc** [comment]
you need to save all the pictures in the project and embed them, not insert them with code. It's not readable and a lot of redundant code

File: src/components/header/components/mobile/Mobile.component.tsx (hunk 7 above)
Comment (3apy6a) [reply src/components/header/components/mobile/Mobile.component.tsx]: done

File: src/components/header/components/shared/Cart.component.tsx (hunk 6 above)
Comment (3apy6a) [reply src/components/header/components/shared/Cart.component.tsx]: done

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**